### DISCUSSION TITLE
Displaying laravel mail notifications

### DISCUSSION THREAD

**Discussion by smknstd** [Xammie/mailbook#29] [Q&A] [answered]
As discussed on twitter it would be great to use with mail notifications https://laravel.com/docs/9.x/notifications#mail-notifications

**Comment by smknstd** [comment]
sorry for duplicate https://github.com/Xammie/mailbook/discussions/27

**Reply by Xammie** [reply]
No problem, I've already looked into it this week. Its a bit tricky and I will need to rewrite some things but it should be possible.

**Reply by smknstd** [reply]
Yeah notifications can definitely be displayed https://laravel.com/docs/9.x/notifications#previewing-mail-notifications

but you will need to mix `\Illuminate\Mail\Mailable` and `\Illuminate\Notifications\Messages\MailMessage`, good luck

**Reply by Xammie** [reply]
Ah interesting, I haven't seen that part of the docs yet.

Although I have a completely different approach I'm trying by actually sending the email and capturing it using a custom email driver. If it works I won't have to deal with 2 different interfaces.

**Reply by smknstd** [reply]
Wow, sounds interesting ! Looking forward to it :) Personally I use mailtrap for testing emails locally, so I'd like to continue using it in addition to mailbook.

https://beyondco.de/docs/helo/getting-started/installation has the same concept but uses local smtp as email driver. Their UI looks really good. Just noticed the "broken link" check, which seems a nice feature

**Reply by Xammie** [reply]
Yeah I use something like that too. Mailbook won't replace your mailtrap or HELO. The plan is that it will inject a driver when it is needed to render a mail and it will not interfere with your preferred mail driver.

I will let you know when I have a proof of concept. My aim is to finish it this week. Will keep you posted :)

**Comment by Xammie** [comment]
@smknstd I've finished the support for notifications and would appreciate if you could test it. Let me know how it goes!

```
composer require xammie/mailbook dev-support-notifications
```

**Reply by smknstd** [reply]
It seems to work great ! I can preview both mails and notifications :)

Just for your information I got an error first because my project uses the old laravel mail config `MAIL_DRIVER` instead of `MAIL_MAILER`.

**Reply by Xammie** [reply]
Thanks a lot of the feedback! Was that by any chance a laravel 8 project?

**Reply by smknstd** [reply]
No it has been migrated to Laravel 9 few months ago but it comes from Laravel 5. The new config appeared with  Laravel 7 and it was labelled as "optional" in the upgrade guide.

https://laravel.com/docs/7.x/upgrade

![Screenshot 2022-12-20 at 12-44-26 Laravel - The PHP Framework For Web Artisans](https://user-images.githubusercontent.com/2412608/208659495-fe0346c2-4008-4706-aa2c-de544f809039.png)

**Comment by Xammie** [comment, marked as answer]
I just released version [1.5.0](https://github.com/Xammie/mailbook/releases/tag/1.5.0) which now supports notifications! 🎉

**Reply by smknstd** [reply]
Excellent ! Will update soon (early 2023 ;). Your work is appreciated here (mailbook already helped to spot and fix some not so responsive old notifications).
